AirCasting/SearchAndFollow/SearchMap/SearchMapView.swift

```diff
@@ -0,0 +1,93 @@
+// Created by Lunar on 16/02/2022.
+//
+
+import CoreLocation
+import Foundation
+import SwiftUI
+import AirCastingStyling
+
+struct SearchMapView: View {
+    @StateObject var viewModel: SearchMapViewModel
+    @Environment(\.presentationMode) var presentationMode
+    
+    init(locationName: String, locationAddress: CLLocationCoordinate2D, parametr: String) {
+        _viewModel = .init(wrappedValue: .init(passedLocation: locationName, passedLocationAddress: locationAddress, passedParameter: parametr))
+    }
+    
+    var body: some View {
+        ZStack(alignment: .top, content: {
+            LoadingView(isShowing: $viewModel.showLoadingIndicator, activityIndicatorText: Strings.SearchMapView.loadingText) {
+                map
+            }
+            VStack(alignment: .center, content: {
+                textField
+                parametersText
+                redoneButton
+            })
+                .padding(.top, 50)
+                .padding(.horizontal)
+        })
+            .onChange(of: viewModel.shouldDismissView, perform: { _ in
+                viewModel.shouldDismissView ? self.presentationMode.wrappedValue.dismiss() : nil
+            })
+            .alert(item: $viewModel.alert, content: { $0.makeAlert() })
+    }
+}
+
+// MARK: - Private View Components
+private extension SearchMapView {
+    var textField: some View {
+        createTextfield(placeholder: "", binding: .constant(viewModel.passedLocation))
+            .disabled(true)
+    }
+    
+    var parametersText: some View {
+        Text(String(format: Strings.SearchMapView.parameterText, arguments: [viewModel.passedParameter]))
+            .font(.muli(size: 16, weight: .semibold))
+    }
+    
+    var redoneButton: some View {
+        Button {
+            withAnimation(.easeOut(duration: 0.2)) {
+                viewModel.redoTapped()
+            }
+        } label: {
+            Text("\(Strings.SearchMapView.redoText) \(Image(systemName: "goforward"))")
+                .font(Fonts.boldHeading3)
+                .lineLimit(1)
+                .scaledToFill()
+                .multilineTextAlignment(.leading)
+        }
+        .foregroundColor(.white)
+        .frame(width: UIScreen.main.bounds.width / 2.2, height: 8, alignment: .center)
```

Comment:
I'm not a fan of relying on `UIScreen` size inside view elements. It makes them unflexible. We won't be able to shrink this view, it will always need to be a full screen width. It's much better to use `GeometryReader`, read the views width and make calculations based on that. For example - if we migrate this view to iPad and we make it half the size of the screen (for example we'll be showing current session data on the right side, idk) the button will be too big because the view won't calculate the width base on the actual view size, instead it will take screen size.

I feel I didn't make myself too clear here 😅 if not ping me and we'll talk about it

Reply:
To be honest I think I understood this, thank you for explaining this I never look at it in this way